relayer/src/main.rs

```diff
@@ -1,3 +1,307 @@
+            explorer_transaction_url: LOCAL_CONF.get("explorer_transaction_url").unwrap(),
+        };
+        let json_rpc_client = network_config.json_rpc_client();
+        json_rpc_client
+    } else {
+        let network_config = rpc_config.networks.get(&network_name).unwrap();
+        let json_rpc_client = network_config.json_rpc_client();
+        json_rpc_client
+    }
+
+});
+static RELAYER_ACCOUNT_ID: Lazy<String> = Lazy::new(|| {
+    let relayer_account_id: String = LOCAL_CONF.get("relayer_account_id").unwrap();
+    relayer_account_id
+});
+static KEYS_FILENAME: Lazy<String> = Lazy::new(|| {
+   let keys_filename: String = LOCAL_CONF.get("keys_filename").unwrap();
+    keys_filename
+});
+static PORT: Lazy<u16> = Lazy::new(|| {
+    let port: u16 = LOCAL_CONF.get("port").unwrap();
+    port
+});
+
+#[tokio::main]
+async fn main() {
+    // initialize tracing (aka logging)
+    tracing_subscriber::registry().with(tracing_subscriber::fmt::layer()).init();
+
+    // build our application with a route
+    let app = Router::new()
+        // `POST /relay` goes to `relay` handler function
+        .route("/relay", post(relay));
+        // See https://docs.rs/tower-http/0.1.1/tower_http/trace/index.html for more details.
+        //.layer(TraceLayer::new_for_http()); // TODO LP: re-add when tower-http dependency conflict is resolved
+
+    // run our app with hyper
+    // `axum::Server` is a re-export of `hyper::Server`
+    let addr = SocketAddr::from(([127, 0, 0, 1], PORT.clone()));
+    info!("listening on {}", addr);
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service())
+        .await
+        .unwrap();
+}
+
+async fn relay(
+    data: extract::Json<Vec<u8>>,
+) -> impl IntoResponse {
+    // deserialize SignedDelegateAction using borsh
+    match SignedDelegateAction::try_from_slice(&data.0) {
+        Ok(signed_delegate_action) => {
+            debug!("Deserialized SignedDelegateAction object: {:#?}", signed_delegate_action);
+
+            // get the latest block hash
+            let latest_final_block_hash = JSON_RPC_CLIENT
+                .call(near_jsonrpc_client::methods::block::RpcBlockRequest{
+                    block_reference: BlockReference::Finality(Finality::Final)
+                }).await.unwrap().header.hash;
+
+            // create Transaction from SignedDelegateAction
+            let unsigned_transaction = Transaction{
+                signer_id: RELAYER_ACCOUNT_ID.as_str().parse().unwrap(),
+                public_key: signed_delegate_action.delegate_action.public_key,
+                nonce: signed_delegate_action.delegate_action.nonce,
+                // the receiver of the txn is the sender of the signed delegate action
+                receiver_id: signed_delegate_action.delegate_action.sender_id,
+                block_hash: latest_final_block_hash,
+                actions: signed_delegate_action.delegate_action.actions
```

Comment:
I believe this line turns the `transaction` with `delegate_action` into a normal `transaction` with `actions` in it. 
`actions: signed_delegate_action.delegate_action.actions` => `actions: vec![signed_delegate_action.delegate_action]`